packages/adminrouter/extra/src/README.md

```diff
@@ -24,9 +24,24 @@ Admin Router runs on both master and agent nodes, each with different configurat
 - HMTL: [docs/api/nginx.agent.html](docs/api/nginx.agent.html)
 - Rendered: <https://rawgit.com/dcos/dcos/master/packages/adminrouter/extra/src/docs/api/nginx.agent.html>
 
-Use `make api-docs` to regenerate the YAML and HTML files.
+## Endpoints documentation
 
-Use `make check-api-docs` to validate that the YAML and HTML files are up to date.
+All Admin Router endpoints are documented using [nginxdox](https://github.com/karlkfi/ngindox)
+tool, which uses special format of the comments in order to describe endpoint
+configuration, and automatically parse it into HTML documents. Please check the
+project documentation for more details.
+
+Admin Router CI automatically checks if the endpoint documentation generated
+using nginxdox and embedded into the repository is up to date. If not, the CI
+job is failed and the user needs to regenerate the docs and re-submit the PR.
+
+The check is done by generating the documentation also during the build stage.
+If, after the nginxdox run, git detects uncommitted changes, then this means
+that the Admin Router configuration differs from the HTML documents that
+are committed into repository. This is done using `make check-api-docs` target.
+
+In order to regenerate the documentation files, one needs to execute
+`make api-docs` file and commit the changes into the repository.
```

Comment:
I guess change "one needs to execute `make...` file`..." with "one needs to execute the`make...` command".